pkg/csi/service/vanilla/controller_helper.go

```diff
@@ -69,3 +55,32 @@ func validateVanillaControllerUnpublishVolumeRequest(ctx context.Context, req *c
 func validateVanillaControllerExpandVolumeRequest(ctx context.Context, req *csi.ControllerExpandVolumeRequest) error {
 	return common.ValidateControllerExpandVolumeRequest(ctx, req)
 }
+
+// registerVolume helps register volume specified by volumePath with CNS
+// function returns volumeID after successful registration
+func registerVolume(ctx context.Context, volumePath string, c *controller) (string, error) {
+	log := logger.GetLogger(ctx)
+	var registeredVolumeID string
+	volumeMigrationSpec, err := volumeMigrationService.GetVolumeInfo(ctx, volumePath)
```

Comment:
GetVolumeInfo can register the volume if not found in the cache. This way we can hide that cache design in volumeMigrationService to callers. In this PR, you are making callers aware of the cache and you are making them responsible for maintaining this cache which can be avoided.

Reply:
We can discuss this. there is a reason we are keeping GetVolumeInfo without register volume. Syncer container has use case. @chethanv28

Reply:
Hide the cache design. Please see - https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/235/commits/6ee465b91b8f7164c77e0abf6b33b5a1f16761f6